packages/themes/build.js

```diff
@@ -0,0 +1,243 @@
+      result[key] = traverseAndReplace(obj[key], oldStr, newStr);
+    });
+    return result;
+  } else {
+    return obj;
+  }
+};
+
+// This function replaces strings in the mode within the semantic key of the object
+const replaceStringInMode = (obj, mode, oldStr, newStr) => {
+  if (typeof obj !== 'object' || obj === null) {
+    return obj;
+  }
+
+  // Replace strings in the specified mode within the semantic key
+  if (obj.semantic && obj.semantic[mode]) {
+    obj.semantic[mode] = traverseAndReplace(obj.semantic[mode], oldStr, newStr);
+  }
+
+  // Recursively apply the function to all keys of the object
+  Object.keys(obj).forEach((key) => {
+    obj[key] = replaceStringInMode(obj[key], mode, oldStr, newStr);
+  });
+
+  return obj;
+};
+
+// This function renames keys in the object based on the keyMap
+const renameKeysDeep = (obj, keyMap) => {
+  if (typeof obj !== 'object' || obj === null) {
+    return obj;
+  }
+
+  if (Array.isArray(obj)) {
+    return obj.map((item) => renameKeysDeep(item, keyMap));
+  }
+
+  const renamedObj = {};
+  Object.keys(obj).forEach((key) => {
+    const newKey = keyMap[key] || key;
+    renamedObj[newKey] = renameKeysDeep(obj[key], keyMap);
+  });
+  return renamedObj;
+};
+
+StyleDictionary.registerParser({
+  pattern: /\.json$/,
+  parse: ({ contents }) => {
+    try {
+      const jsonObj = JSON.parse(contents);
+
+      // Replace the references to modes
+      const lightParsed = replaceStringInMode(
+        jsonObj['variables'],
+        'light-mode',
+        'semantic.dark-mode', // Reference to incorrect mode prior to replacement
+        'theme.light'
+      );
+
+      const darkParsed = replaceStringInMode(
+        lightParsed,
+        'dark-mode',
+        'semantic.dark-mode',
+        'theme.dark'
+      );
+
+      const parsedJsonObj = renameKeysDeep(darkParsed, {
+        semantic: 'theme',
+        'light-mode': 'light',
+        'dark-mode': 'dark'
+      });
+
+      return parsedJsonObj;
+    } catch (error) {
+      console.error(error);
+    }
+  }
+});
+
+console.log('Build started...');
+console.log('\n==============================================');
+
+/*
+ THESE ARE CONTITIONS TO FILTER ON AND A HELPER FUNCTION TO CREATE MATCHING FILTERS
+ These are a convenience to avoid long filter strings such as
+ token.attributes.category === 'global' || token.attributes.category === 'spacing' || token....
+ The objects defined here are allowed through - everything else is filtered out. Acts as a yes gate
+ The not: array allows for partial filtering of object keys. Acts as a no gate
+*/
+const conditionsCssLightTheme = [{ category: 'theme', type: 'light' }];
+
+const conditionsCssDarkTheme = [{ category: 'theme', type: 'dark' }];
+
+const conditionsCssGlobal = [
+  // token.attributes.category === 'global' || token.attributes.category === 'spacing'
+  { category: 'global', subitem: { not: ['seed'] } },
+  { category: 'spacing' }
+];
+
+const conditionsTw = [
```

Comment:
I'd rename this to `conditionsTailwind`: it is more explicit, and not too long.

Reply:
ok